### PR TITLE
Reworking "Bugfix/mongodb unsupported char" (pull request #113)

### PR DESCRIPTION
**Pull request by leodutra**
Removes unsupported chars from collection names.
**Introduces a flag (`useRawColletionNames`) for easy fallback**, since it is a breaking change.

Enhances [Massot's sanitization](https://github.com/OptimalBits/node_acl/pull/113) by **replacing slashes and spaces by underlines**, easing reading. 
Merges the old pull request code with the latest changes from the master branch of this repository, without conflicts.

Please, review.

### CODE REVIEW

**Comment by manast** [comment]
Thanks for the patch. I will review it as soon as I can.

**Comment by manast** [comment]
I am wondering, is not that when using invalid collection characters, mongodb will complain? if so, in which case would you like to use raw collection names?

**Comment by leodutra** [comment]
The problem is with mongodump, a common tool most of us use. It cannot
create dumps with slashes cause of file system restrictions. Is not a bug,
but a really bad practice to allow anything.

Refer to pull request 113 for more om this.

Raw flag is intended for easy migration and turning feature off if really
required.

**Comment by leodutra** [comment]
Reminding. 

If this become old, conflicts can happen with newer code changes and it will have to be reworked.

File: lib/mongodb-backend.js [hunk 1]
```diff
@@ -199,11 +200,10 @@ MongoDBBackend.prototype = {
   }
 }
 
-function removeUnsupportedChar(text) {
-  if (typeof text == 'string' || text instanceof String) {
+function removeUnsupportedChar(text, useRawColletionNames) {
```
Comment (leodutra) [inline lib/mongodb-backend.js]: Lol... you are right.

Comment (leodutra) [inline lib/mongodb-backend.js]: No problem in it being exposed? Use '_' (internal) prefix?

**Comment by leodutra** [comment]
Review changes done. Test needed. I don't have a test environment on my current machine.
Please, test before pulling or wait for my test.

**Comment by rxjs-space** [comment]
@manast , Hi, the npm package is not updated to include this pr. Still having trouble as #53.

**Pull request merged**